src/rulesets/Base/fastmath_able.jl

```diff
@@ -52,7 +52,23 @@ let
         # exponents
         @scalar_rule cbrt(x) inv(3 * Ω ^ 2)
         @scalar_rule inv(x) -(Ω ^ 2)
-        @scalar_rule sqrt(x) inv(2Ω)  # gradient +Inf at x==0
+        # ensure that at sqrt(0), a zero (co)tangent produces a zero (co)tangent
+        function frule((_, Δx), ::typeof(sqrt), x::Number)
+            Ω = sqrt(x)
+            ∂Ω = Δx / 2Ω
+            return Ω, ifelse(iszero(Δx) & iszero(x), zero(∂Ω), ∂Ω)
```

Comment:
Is there a specific reason to use `ifelse` instead of a ternary operator (which does not require to evaluate both branches)?

Reply:
My reasoning was that so long as the type of `∂Ω` is inferrable, the two branches do no extra work, and the use of `&` and `ifelse` both could perform better if this is used in an inner loop and potentially allow Zygote to perform better for higher order AD (since Zygote tends so be slow when hitting control flow but has a special rule for `ifelse`). However, I was unable to devise a benchmark that showed a substantial difference.

Reply:
I know that in older Julia versions there were cases where we could improve performance in SciML by avoiding `zero` or moving it out of loops. But I couldn't reproduce this with a simple example immediately, maybe it's not relevant here and/or fixed in recent Julia versions.

Reply:
An example where it matters:
```julia
julia> using BenchmarkTools

julia> function f(x)    
           s = zero(x)      
           for i in 1:10    
               s += iseven(i) ? zero(x) : x
           end   
           return s
       end             
f (generic function with 1 method) 
                                                     
julia> function g(x)       
           s = zero(x)      
           for i in 1:10
               s += ifelse(iseven(i), zero(x), x)
           end
           return s
       end                 
g (generic function with 1 method)

julia> @btime f($(big"1.0"));
  45.640 μs (3002 allocations: 164.17 KiB)

julia> @btime g($(big"1.0"));
  56.341 μs (4002 allocations: 218.86 KiB)
``